### PR TITLE
Add xvfb to requirements.

### PR DESCRIPTION
**Pull request by brettle**
**Description**
The build of the snap fails on a stock bionic system because xvfb is not installed.

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
Thank you!

But in this case `xvfb` is only needed to create the snap and for running Webots on native systems, so we don't want to add it as a runtime dependency of Webots.
This should be added in the [webots-snap](https://github.com/cyberbotics/webots-snap) repo.

**Comment by brettle** [comment]
Since `make install` fails on a stock headless bionic system, I'd argue this is a bug in this repo not `webots-snap`. I agree it's a build dependency, not a runtime dependency though.

**Comment by stefaniapedrazzi** [comment]
Here are the instruction for example for installing Webots on Docker:
https://github.com/cyberbotics/webots/wiki/Docker#get-and-compile-webots
where it is specified that `xvfb` has to be installed.
But this is only required for running Webots on headless systems.
It is a very particular case and we don't want to install xvfb on other systems using the default installation scripts.

We can evaluate if it makes sense to add an additional script for headless system, but it will contain only `xvfb` for the moment, so I'm not sure it makes much sense.

**Comment by stefaniapedrazzi** [comment]
I can add it in the Linux installation page:
https://github.com/cyberbotics/webots/wiki/Linux-installation

**Comment by brettle** [comment]
If it's added in the compilation dependencies script, it won't be included in the deb package, right? It will just be installed for people trying to build from source. The addition would simplify the build documentation. I'm really not seeing the downside.

**Comment by stefaniapedrazzi** [comment]
The downside is that it will be installed on the computer of all the users and developers that are compiling Webots and have a monitor and won't need the `xvfb` package on their system.

**Comment by brettle** [comment]
It's only 2.2MB which is a drop in the bucket compared to all of the other deps that need to be installed to build from source. Weigh that against the time and frustration of developers who need it but don't have it installed. For example, it has taken you multiple attempts to get [the new automated snap build](https://github.com/cyberbotics/webots-snap/pull/23) to work because of this issue. That would not have occurred if this had been fixed.

**Comment by stefaniapedrazzi** [comment]
Yes, in https://github.com/cyberbotics/webots-snap/pull/23 I'm experimenting but it is clear from the error message that `xvfb` is causing issues.

I got your point and I'm taking seriously into consideration your use case. But we should also consider all the other use cases. For example to run Webots in headless system it is not enough to install `xvfb` but it also needs to be setup before being able to start Webots.

So we think that the best option is, in the runtime dependencies script, to check if the system is headless, for example checking if the `DISPLAY` environment variable is empty, and if yes install `xvfb`.

A second step, as much importart as this one, would be to automatically setup and start `xvfb` and the other variables if the DISPLAY variable is empty (headless mode) so that Webots can be started out of the box.

**Comment by brettle** [comment]
Sounds reasonable. I've made the suggested changes to the scripts and packaging Makefile. I'm assuming that getting webots to start out of the box on a headless server will be addressed separately.

File: scripts/packaging/Makefile [hunk 1]
```diff
@@ -104,8 +104,10 @@ iscc_formatter.exe: iscc_formatter.c
 	+@$(CC) -Wall -o $@ iscc_formatter.c
 
 xvfb:
+ifndef DISPLAY
 	+@echo "# starting Xvfb: X Virtual Frame Buffer"
 	+@Xvfb :99 -screen 0 1024x768x16 &
+endif
```
Comment (stefaniapedrazzi) [inline scripts/packaging/Makefile:110]: See previous comment
```suggestion
```

**Review by stefaniapedrazzi — CHANGES_REQUESTED** [review]
There are some syntax errors making the tests crashing and a slightly different approach should be used in the packaging Makefile to not break the automatic generation of snap packages.

**Review by stefaniapedrazzi — APPROVED** [review]
Thank you!

I restarted the tests that were failing due to issues with in some dependency and now the Linux tests passed successfully.

**Pull request merged**